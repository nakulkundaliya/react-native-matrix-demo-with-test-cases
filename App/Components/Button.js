import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import { TouchableOpacity, Text } from 'react-native';
import styles from './Styles/ButtonStyle';

const Button = props => {
  const { name, onPress, style, disabled } = props;
  return (
    <TouchableOpacity
      style={[styles.buttonStyle, style]}
      onPress={() => onPress()}
      disabled={disabled}
    >
      <Text style={styles.buttonTextStyle}>{name}</Text>
    </TouchableOpacity>
  );
};
export default Button;
