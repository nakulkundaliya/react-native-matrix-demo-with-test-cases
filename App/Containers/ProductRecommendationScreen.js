import React, { Component } from 'react';
import { ScrollView, View, TextInput, Text } from 'react-native';
import { connect } from 'react-redux';
import RadioForm from 'react-native-simple-radio-button';
import Button from '../Components/Button';
import bundles from '../Fixtures/bundles.json';
// Add Actions - replace 'Your' with whatever your reducer is called :)
// import YourActions from '../Redux/YourRedux'

// Styles
import styles from './Styles/ProductRecommendationScreenStyle';
const RADIO_PARAMS = [
  { label: 'Yes ', value: 'Yes' },
  { label: 'No', value: 'No' }
];

class ProductRecommendationScreen extends Component {
  static navigationOptions = {
    title: 'Product Recommendation'
  };

  constructor(props) {
    super(props);
    this.state = {
      age: '',
      isStudent: 'Yes',
      income: '',
      bundles,
      bundle: null
    };
  }

  submit = () => {
    const {
      age,
      income,
      isStudent,
      bundles: { juniorSaver, student, classic, classicPlus, gold }
    } = this.state;
    const disable = isNaN(age) || isNaN(income) || !age || !income;

    if (disable) {
      alert('Pleaae enter proper value');
      return;
    }
    if (age < 18) {
      this.setState({
        bundle: juniorSaver
      });
    } else if (isStudent === 'Yes') {
      this.setState({
        bundle: student
      });
    } else if (income > 0 && income <= 12000) {
      this.setState({
        bundle: classic
      });
    } else if (income > 12000 && income <= 40000) {
      this.setState({
        bundle: classicPlus
      });
    } else if (income > 40000) {
      this.setState({
        bundle: gold
      });
    }
  };

  updateFormValue = (key, value) => {
    this.setState({ [key]: value });
  };

  renderForm() {
    const { age, income, isStudent } = this.state;
    const disable = !age || !income || !isStudent;
    return (
      <View>
        <TextInput
          placeholder="Enter age"
          keyboardType="number-pad"
          style={styles.textInputStyle}
          onChangeText={value => this.updateFormValue('age', value)}
          value={age}
        />
        <View>
          <Text style={styles.studentTextStyle}>Are you student ?</Text>
          <RadioForm
            radio_props={RADIO_PARAMS}
            initial={0}
            formHorizontal={true}
            labelHorizontal={true}
            buttonColor={'#2196f3'}
            animation={true}
            onPress={value => this.updateFormValue('isStudent', value)}
          />
        </View>

        <TextInput
          placeholder="Enter Income"
          keyboardType="number-pad"
          style={styles.textInputStyle}
          onChangeText={value => this.updateFormValue('income', value)}
          value={income}
        />
        <Button
          name="Submit"
          onPress={this.submit}
          style={{ width: '100%' }}
          disable={disable}
        />
      </View>
    );
  }

  renderOutput() {
    const { bundle } = this.state;
    if (!bundle) return;
    return (
      <View>
        <Text style={{ fontWeight: '800', fontSize: 18 }}>Bundle Output</Text>
        <Text>Bundle Name : {bundle.name}</Text>
        <Text>Products : {bundle.products.join(',')}</Text>
        <Text>Value: {bundle.value}</Text>
      </View>
    );
  }

  render() {
    return (
      <ScrollView style={styles.container}>
        {this.renderForm()}
        {this.renderOutput()}
      </ScrollView>
    );
  }
}

export default ProductRecommendationScreen;
