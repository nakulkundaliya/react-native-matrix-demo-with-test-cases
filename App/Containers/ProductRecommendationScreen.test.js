import React from 'react';
import { shallow, mount } from 'enzyme';
import ProductRecommendationScreen from './ProductRecommendationScreen';
import Button from '../Components/Button';

describe('ProductRecommendationScreen', () => {
  describe('Rendering', () => {
    let component;

    beforeEach(() => {
      component = shallow(<ProductRecommendationScreen />);
    });

    it('should be defined', () => {
      expect(ProductRecommendationScreen).toBeDefined();
    });

    it('should match to snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('should check form validation', () => {
      component.setState({ age: '', isStudent: '', income: '' });
      alert = jest.fn();
      const instance = component.instance();
      instance.updateFormValue('isStudent', 'No');
      instance.updateFormValue('income', '10000');
      component
        .find(Button)
        .first()
        .props()
        .onPress();
      expect(alert).toBeCalled();
    });

    it('should generate bundle output', () => {
      const instance = component.instance();
      instance.updateFormValue('age', '22');
      instance.updateFormValue('isStudent', 'No');
      instance.updateFormValue('income', '20000');
      jest.spyOn(instance, 'submit');

      component
        .find(Button)
        .first()
        .props()
        .onPress();
      expect(instance.state.bundle.value).toBe(2);
    });
  });
});
