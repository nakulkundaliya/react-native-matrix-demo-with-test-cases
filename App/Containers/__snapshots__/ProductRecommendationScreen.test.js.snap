// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRecommendationScreen Rendering should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductRecommendationScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View>
          <TextInput
            allowFontScaling={true}
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="Enter age"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "#595959",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 50,
                "justifyContent": "center",
                "marginVertical": 10,
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View>
            <Text
              style={
                Object {
                  "marginVertical": 5,
                }
              }
            >
              Are you student ?
            </Text>
            <RadioForm
              animation={true}
              buttonColor="#2196f3"
              disabled={false}
              formHorizontal={true}
              initial={0}
              labelColor="#000"
              labelHorizontal={true}
              onPress={[Function]}
              radio_props={
                Array [
                  Object {
                    "label": "Yes ",
                    "value": "Yes",
                  },
                  Object {
                    "label": "No",
                    "value": "No",
                  },
                ]
              }
              selectedButtonColor="#2196f3"
              selectedLabelColor="#000"
              wrapStyle={Object {}}
            />
          </View>
          <TextInput
            allowFontScaling={true}
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="Enter Income"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "#595959",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 50,
                "justifyContent": "center",
                "marginVertical": 10,
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <Button
            disable={true}
            name="Submit"
            onPress={[Function]}
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </View>,
        undefined,
      ],
      "style": Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "flex": 1,
        "marginHorizontal": 15,
        "paddingTop": 10,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TextInput
              allowFontScaling={true}
              keyboardType="number-pad"
              onChangeText={[Function]}
              placeholder="Enter age"
              rejectResponderTermination={true}
              style={
                Object {
                  "borderColor": "#595959",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />,
            <View>
              <Text
                style={
                  Object {
                    "marginVertical": 5,
                  }
                }
              >
                Are you student ?
              </Text>
              <RadioForm
                animation={true}
                buttonColor="#2196f3"
                disabled={false}
                formHorizontal={true}
                initial={0}
                labelColor="#000"
                labelHorizontal={true}
                onPress={[Function]}
                radio_props={
                  Array [
                    Object {
                      "label": "Yes ",
                      "value": "Yes",
                    },
                    Object {
                      "label": "No",
                      "value": "No",
                    },
                  ]
                }
                selectedButtonColor="#2196f3"
                selectedLabelColor="#000"
                wrapStyle={Object {}}
              />
            </View>,
            <TextInput
              allowFontScaling={true}
              keyboardType="number-pad"
              onChangeText={[Function]}
              placeholder="Enter Income"
              rejectResponderTermination={true}
              style={
                Object {
                  "borderColor": "#595959",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />,
            <Button
              disable={true}
              name="Submit"
              onPress={[Function]}
              style={
                Object {
                  "width": "100%",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "keyboardType": "number-pad",
              "onChangeText": [Function],
              "placeholder": "Enter age",
              "rejectResponderTermination": true,
              "style": Object {
                "borderColor": "#595959",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 50,
                "justifyContent": "center",
                "marginVertical": 10,
                "paddingHorizontal": 10,
                "width": "100%",
              },
              "underlineColorAndroid": "transparent",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  style={
                    Object {
                      "marginVertical": 5,
                    }
                  }
                >
                  Are you student ?
                </Text>,
                <RadioForm
                  animation={true}
                  buttonColor="#2196f3"
                  disabled={false}
                  formHorizontal={true}
                  initial={0}
                  labelColor="#000"
                  labelHorizontal={true}
                  onPress={[Function]}
                  radio_props={
                    Array [
                      Object {
                        "label": "Yes ",
                        "value": "Yes",
                      },
                      Object {
                        "label": "No",
                        "value": "No",
                      },
                    ]
                  }
                  selectedButtonColor="#2196f3"
                  selectedLabelColor="#000"
                  wrapStyle={Object {}}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Are you student ?",
                  "style": Object {
                    "marginVertical": 5,
                  },
                },
                "ref": null,
                "rendered": "Are you student ?",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "animation": true,
                  "buttonColor": "#2196f3",
                  "disabled": false,
                  "formHorizontal": true,
                  "initial": 0,
                  "labelColor": "#000",
                  "labelHorizontal": true,
                  "onPress": [Function],
                  "radio_props": Array [
                    Object {
                      "label": "Yes ",
                      "value": "Yes",
                    },
                    Object {
                      "label": "No",
                      "value": "No",
                    },
                  ],
                  "selectedButtonColor": "#2196f3",
                  "selectedLabelColor": "#000",
                  "wrapStyle": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "keyboardType": "number-pad",
              "onChangeText": [Function],
              "placeholder": "Enter Income",
              "rejectResponderTermination": true,
              "style": Object {
                "borderColor": "#595959",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 50,
                "justifyContent": "center",
                "marginVertical": 10,
                "paddingHorizontal": 10,
                "width": "100%",
              },
              "underlineColorAndroid": "transparent",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disable": true,
              "name": "Submit",
              "onPress": [Function],
              "style": Object {
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      undefined,
    ],
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View>
            <TextInput
              allowFontScaling={true}
              keyboardType="number-pad"
              onChangeText={[Function]}
              placeholder="Enter age"
              rejectResponderTermination={true}
              style={
                Object {
                  "borderColor": "#595959",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View>
              <Text
                style={
                  Object {
                    "marginVertical": 5,
                  }
                }
              >
                Are you student ?
              </Text>
              <RadioForm
                animation={true}
                buttonColor="#2196f3"
                disabled={false}
                formHorizontal={true}
                initial={0}
                labelColor="#000"
                labelHorizontal={true}
                onPress={[Function]}
                radio_props={
                  Array [
                    Object {
                      "label": "Yes ",
                      "value": "Yes",
                    },
                    Object {
                      "label": "No",
                      "value": "No",
                    },
                  ]
                }
                selectedButtonColor="#2196f3"
                selectedLabelColor="#000"
                wrapStyle={Object {}}
              />
            </View>
            <TextInput
              allowFontScaling={true}
              keyboardType="number-pad"
              onChangeText={[Function]}
              placeholder="Enter Income"
              rejectResponderTermination={true}
              style={
                Object {
                  "borderColor": "#595959",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <Button
              disable={true}
              name="Submit"
              onPress={[Function]}
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </View>,
          undefined,
        ],
        "style": Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "flex": 1,
          "marginHorizontal": 15,
          "paddingTop": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TextInput
                allowFontScaling={true}
                keyboardType="number-pad"
                onChangeText={[Function]}
                placeholder="Enter age"
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#595959",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": 50,
                    "justifyContent": "center",
                    "marginVertical": 10,
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />,
              <View>
                <Text
                  style={
                    Object {
                      "marginVertical": 5,
                    }
                  }
                >
                  Are you student ?
                </Text>
                <RadioForm
                  animation={true}
                  buttonColor="#2196f3"
                  disabled={false}
                  formHorizontal={true}
                  initial={0}
                  labelColor="#000"
                  labelHorizontal={true}
                  onPress={[Function]}
                  radio_props={
                    Array [
                      Object {
                        "label": "Yes ",
                        "value": "Yes",
                      },
                      Object {
                        "label": "No",
                        "value": "No",
                      },
                    ]
                  }
                  selectedButtonColor="#2196f3"
                  selectedLabelColor="#000"
                  wrapStyle={Object {}}
                />
              </View>,
              <TextInput
                allowFontScaling={true}
                keyboardType="number-pad"
                onChangeText={[Function]}
                placeholder="Enter Income"
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#595959",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": 50,
                    "justifyContent": "center",
                    "marginVertical": 10,
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />,
              <Button
                disable={true}
                name="Submit"
                onPress={[Function]}
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": true,
                "keyboardType": "number-pad",
                "onChangeText": [Function],
                "placeholder": "Enter age",
                "rejectResponderTermination": true,
                "style": Object {
                  "borderColor": "#595959",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                },
                "underlineColorAndroid": "transparent",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Text
                    style={
                      Object {
                        "marginVertical": 5,
                      }
                    }
                  >
                    Are you student ?
                  </Text>,
                  <RadioForm
                    animation={true}
                    buttonColor="#2196f3"
                    disabled={false}
                    formHorizontal={true}
                    initial={0}
                    labelColor="#000"
                    labelHorizontal={true}
                    onPress={[Function]}
                    radio_props={
                      Array [
                        Object {
                          "label": "Yes ",
                          "value": "Yes",
                        },
                        Object {
                          "label": "No",
                          "value": "No",
                        },
                      ]
                    }
                    selectedButtonColor="#2196f3"
                    selectedLabelColor="#000"
                    wrapStyle={Object {}}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Are you student ?",
                    "style": Object {
                      "marginVertical": 5,
                    },
                  },
                  "ref": null,
                  "rendered": "Are you student ?",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "animation": true,
                    "buttonColor": "#2196f3",
                    "disabled": false,
                    "formHorizontal": true,
                    "initial": 0,
                    "labelColor": "#000",
                    "labelHorizontal": true,
                    "onPress": [Function],
                    "radio_props": Array [
                      Object {
                        "label": "Yes ",
                        "value": "Yes",
                      },
                      Object {
                        "label": "No",
                        "value": "No",
                      },
                    ],
                    "selectedButtonColor": "#2196f3",
                    "selectedLabelColor": "#000",
                    "wrapStyle": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": true,
                "keyboardType": "number-pad",
                "onChangeText": [Function],
                "placeholder": "Enter Income",
                "rejectResponderTermination": true,
                "style": Object {
                  "borderColor": "#595959",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                },
                "underlineColorAndroid": "transparent",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disable": true,
                "name": "Submit",
                "onPress": [Function],
                "style": Object {
                  "width": "100%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        undefined,
      ],
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
